internal/internal_task_pollers.go

```diff
@@ -746,22 +747,25 @@ func newGetHistoryPageFunc(
 			metricsScope.Counter(metrics.WorkflowGetHistoryFailedCounter).Inc(1)
 			return nil, nil, err
 		}
-
 		metricsScope.Counter(metrics.WorkflowGetHistorySucceedCounter).Inc(1)
 		metricsScope.Timer(metrics.WorkflowGetHistoryLatency).Record(time.Now().Sub(startTime))
-		h := resp.History
-		size := len(h.Events)
-		if size > 0 && atDecisionTaskCompletedEventID > 0 &&
-			h.Events[size-1].GetEventId() > atDecisionTaskCompletedEventID {
-			first := h.Events[0].GetEventId() // eventIds start from 1
-			h.Events = h.Events[:atDecisionTaskCompletedEventID-first+1]
-			if h.Events[len(h.Events)-1].GetEventType() != s.EventTypeDecisionTaskCompleted {
-				return nil, nil, fmt.Errorf("newGetHistoryPageFunc: atDecisionTaskCompletedEventID(%v) "+
-					"points to event that is not DecisionTaskCompleted", atDecisionTaskCompletedEventID)
+		h, err1 := util.DeSerializeBlobDataToHistoryEvents(resp.RawHistory, s.HistoryEventFilterTypeAllEvent)
+		if err1 == nil {
```

Comment:
if err1 != nil, the code will jump to L768 and return nil? Doesn't look right to me. 

In general, we follow the pattern where we first check if err is nil or not and then perform operations on the result.